mintapi/api.py

```diff
@@ -331,38 +333,68 @@ def initiate_account_refresh(email, password):
-    cmdline.add_option('--transactions', '-t', action='store_true', default=False, help='Retrieve transactions')
-    cmdline.add_option('--filename', '-f', help='write results to file. can be {csv,json} format. default is to write to stdout.')
-    cmdline.add_option('--user', '-u', help='mint email login. uses OS keyring to store password info.')
+    try:
+        import keyring
+    except ImportError:
+        keyring = None
 
-    (options, args) = cmdline.parse_args()
+    # Parse command-line arguments {{{
+    cmdline = argparse.ArgumentParser()
+    cmdline.add_argument('email', nargs='?', default=None,
+                         help='The e-mail address for your Mint.com account')
+    cmdline.add_argument('password', nargs='?', default=None,
+                         help='The password for your Mint.com account')
+    cmdline.add_argument('--accounts', action='store_true', dest='accounts',
+                         default=False, help='Retrieve account information'
+                         ' (default if nothing else is specified)')
+    cmdline.add_argument('--budgets', action='store_true', dest='budgets',
+                         default=False, help='Retrieve budget information')
+    cmdline.add_argument('--extended-accounts', action='store_true',
+                         dest='accounts_ext', default=False,
+                         help='Retrieve extended account information (slower, '
+                         'implies --accounts)')
+    cmdline.add_argument('--transactions', '-t', action='store_true',
+                         default=False, help='Retrieve transactions')
+    cmdline.add_argument('--filename', '-f', help='write results to file. can '
+                         'be {csv,json} format. default is to write to '
+                         'stdout.')
+    cmdline.add_argument('--keyring', action='store_true',
+                         help='Use OS keyring for storing password '
+                         'information')
+
+    options = cmdline.parse_args()
+
+    if options.keyring and not keyring:
+        cmdline.error('--keyring can only be used if the `keyring`'
+                      'library is installed.')
 
-    # Handle Python 3's raw_input change.
     try:
-        input = raw_input
+        from __builtin__ import raw_input as input
     except NameError:
         pass
 
-    if options.user:
-        # handle user and password through keyring
-        email = options.user
+    # Try to get the e-mail and password from the arguments
+    email = options.email
+    password = options.password
+
+    if not email:
+        # If the user did not provide an e-mail, prompt for it
+        email = input("Mint e-mail: ")
+
+    if keyring and not password:
+        # If the keyring module is installed and we don't yet have
+        # a password, try prompting for it
         password = keyring.get_password('mintapi', email)
```

Comment:
I think this does what you're looking for now.

Basically, specify `--keyring` (369) but its not installed, you get an error. However, if you don't specify `--keyring` and it IS installed, this code will try and retrieve your password from the keyring if the code hasn't already discovered a password.